lib/ecto/adapters/postgres/connection.ex

```diff
@@ -414,12 +414,11 @@ if Code.ensure_loaded?(Postgrex) do
     defp operator_to_boolean(:and), do: " AND "
     defp operator_to_boolean(:or), do: " OR "
 
-    # Don't wrap json functions in parentheses
-    defp paren_expr({:fragment, _, [{:raw, raw_expr} | _]} = expr, sources, query) do
-      if String.match?(raw_expr, ~r/^\s*jsonb?/i) do
-        [expr(expr, sources, query)]
+    defp parens_for_select([first_expr | _] = expr) do
+      if String.match?(first_expr, ~r/SELECT/i) do
```

Comment:
Could `String.starts_with?(first_expr, "SELECT")` work here?

Reply:
I am not sure, it would then become case sensitive. Also, I think it would break the resulting SQL if somebody accidentally left whitespace at the beginning of the fragment string, e.g. `fragment(" SELECT ...")`.

Reply:
Let's check for select and SELECT. The issue with a regex is also that the behaviour changes if you accidentally have a select in it. For example, if you have a select inside a string.